File: Medium/11. Container With Most Water/ContainerWithMostWater.py
'''
LeetCode Medium: 11. Container With Most Water
https://leetcode.com/problems/container-with-most-water/description/

Check ContainerWithMostWater.cpp for brief solution explanation.
'''
class Solution:
    def maxArea(self, height: List[int]) -> int:
        N = len(height)
        ans, l, r = 0, 0, N-1
        while l<=r:
            ans = max(ans, min(height[l], height[r])*(r-l))
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return ans

File: Medium/11. Container With Most Water/ContainerWithMostWater.java
/*
LeetCode Medium: 11. Container With Most Water
https://leetcode.com/problems/container-with-most-water/description/

Check ContainerWithMostWater.cpp for brief solution explanation.
*/
class Solution {
    public int maxArea(int[] height) {
        int N = height.length;
        int ans = 0, l = 0, r = N-1;
        while(l<=r){
            ans = Math.max(ans, Math.min(height[l], height[r])*(r-l));
            if(height[l]<height[r]){
                l++;
            }else{
                r--;
            }
        }
        return ans;
    }
}